File: week02/Jisoo/03_기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        
        int[] arr = new int[progresses.length];
        for(int i=0; i<progresses.length; i++){
            if ((100 - progresses[i]) % speeds[i] == 0){
                arr[i] = (100 - progresses[i]) / speeds[i];
            } else {
                arr[i] =  (100 - progresses[i]) / speeds[i] + 1;   
            }
        }
        
        int idx = 0;
        int cnt = 1;
        List<Integer> list = new ArrayList<>();
        for(int i=1; i<arr.length; i++){
            if(arr[idx] >= arr[i]){
                cnt++;
            } else {
                list.add(cnt);
                idx = i;
                cnt = 1;
            }
            
        }
        list.add(cnt);
        
        int[] answer = new int[list.size()];
        for(int i=0; i< list.size(); i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}

File: week02/sujin/03_기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    
    remain = []
    for i in range(len(progresses)):
        remain.append(math.ceil((100-progresses[i]) / speeds[i]))
        
    max_days = remain[0]
    count = 1
    for i in range(1, len(remain)):
        
        if max_days < remain[i]:
            max_days = remain[i]
            answer.append(count)
            count = 1
        else:
            count += 1
    answer.append(count)
            
    
    print(remain)
    
    return answer